lib/active_snapshot/models/snapshot.rb

```diff
@@ -145,5 +145,80 @@ def fetch_reified_items(readonly: true)
       return [reified_parent, reified_children_hash]
     end
 
+    def diff(new_snapshot)
```

Comment:
I dont like the idea of making `build_snapshot!` a public method (just to create a snapshot off of the current instance). 

Maybe we could make it a class method on `Snapshot` which accepts arguments that are either a Snapshot instance OR a model instance

```ruby
def self.diff(from:, to:)
   # ...
end
```

Should raise error unless at least one argument is a Snapshot instance

Should also raise error if `from` or `to` are not the same model class and instance ID.